src/convolutions/convolutions.jl

```diff
@@ -0,0 +1,67 @@
+using Base.Cartesian
+
+export Lanczos
+
+"""
+    Lanczos(a=4)
+
+Lanczos resampling via a kernel with size `a`.
+
+This form of interpolation is merely the discrete convolution of the samples with a Lanczos kernel of size `a`. The size is directly related to how "far" the interpolation will reach for information, and has `O(a^2)` impact on runtime. A default value of 4 matches the OpenCV implementation `lanczos4`.
+"""
+struct Lanczos <: InterpolationType
+    a::Integer
```

Comment:
There may also be an opportunity to memoize or otherwise cache the kernel here.

Reply:
According to a few sources I've came across the Lanczos window is defined explicitly with a single parameter ([here is the paper I have been referencing the most, see section 11](http://www.ipol.im/pub/art/2011/g_lmii/article.pdf)). None of the BSplines separate their degree and support, either, so this seems divergent from the current style.

Not saying we couldn't, but it seems like unnecessary work (especially if there's no proper use-case for it).

Reply:
Mathematically lanczos definitely has a single parameter. The practical computation requires a cutoff somewhere though, which is a practical approximation to when the kernel is "almost" zero.  If you plot out the kernel you will notice that it actually is not zero beyond the default limits.

The frequency domain implications of such a cutoff are actually quite profound. The zero cutoff is equivalent to multiplying the real domain by a `rect` which in terms equates to convolution with a sinc.

The BSplines have an explicit cutoff mathematically because they actually are zero beyond a certain point. This is not necessarily the case for lanczos. `sinc(x) * sinc(x/a)` is not zero for `abs(x) >= a`.

If one parameter is given, we can practically revert to the current case. The only real cost here is storing an extra integer. Being able to change the support though will allow one to compare two kernels with distinct scaling factors but the same support.

Reply:
Thanks for the further explanation, I'll add it in!

Reply:
Actually, upon further thinking, I want to be careful about how this is declared.

Since you've written this comment, I've switched to using `Lanczos{a}()` to represent the type, with the expectation that we could use dispatch like `f(::Lanczos{4})` when applicable for specializing fucntions using look up tables, for instance.

If we separate the parameters, we either have to use both as type parameters for allowing specialization, or we commit to not specializing at all (especially because the LUT will be for the default case of same scale and support).

For now, I'll write up an impl that doesn't use any special type parameters or generated functions.